proto/workflow_execution.proto

```diff
@@ -8,58 +8,118 @@ import "google/api/annotations.proto";
 import "google/protobuf/struct.proto";
 import "task_execution.proto";
 
+//available workflow types supported by this WES
+enum WorkflowType {
+    UNKNOWN = 0;
+    CWL = 1;
+    WDL = 2;
+}
+
+//Return envelope for workflow listing
+message WorkflowListResponse {
+   repeated JobDesc workflows = 1;
+   string nextPageToken = 2;
+}
+
+//Small description of workflows, returned by server during listing
+message JobDesc {
+  //REQUIRED
+  string workflow_id  = 1;
+  //REQUIRED
+  ga4gh_task_exec.State state = 2;
+}
 
+//workflow request object
 message WorkflowRequest {
-  string workflow_url = 1;
-  google.protobuf.Struct input = 2;
+  //OPTIONAL
+  //user name of the workflow
+  string name = 1;
+  //OPTIONAL
+  //parameter for execution engine to define/store group information
+  string project_id = 2;
+  //OPTIONAL
+  //free text description of task
+  string description = 3;
```

Comment:
In general I prefer these sorts of not-used-by-the-engine bits of arbitrary metadata to not be in the APIs (see my complaints in [a TES PR](https://github.com/ga4gh/task-execution-schemas/pull/21) for another example).

I feel that they muddy the waters and when one is trying to make a generic API there's a serious risk of overfitting one's vision of how it might be used. Further, it seems like a job for the calling infrastructure. 

If there is a good use case for description or metadata (or even the workflow name, although i feel a lot less strongly about that one) in a manner which is being directly acted upon by a workflow engine I'll retract my stance, but I'm not keen on encoding a "here, hold my stuff" portion to the API.